packages/mui-base/src/ModalUnstyled/ModalUnstyled.tsx

```diff
@@ -249,7 +274,7 @@ const ModalUnstyled = React.forwardRef(function ModalUnstyled(props, ref) {
     ownerState,
   });
 
-  const BackdropComponent = components.Backdrop;
+  const BackdropComponent = components?.Backdrop || 'div';
```

Comment:
Why the `|| div` is added? This changes the default behavior and makes the backdrop always visible.

Reply:
`useSlotProps` expects `elementType` not to be undefined, so I added this as a default that seemed reasonable.
After checking other usages, maybe this would be better?

```suggestion
  const BackdropComponent = components?.Backdrop ?? 'div';
```
If not, I need some Help resolving this.

Reply:
In this case, it seems it's necessary to extend useSlotProps with the support for `undefined` component types. It should return an empty object then. This could be done in a separate PR to keep things focused. Would you like to work on it as well?

Reply:
Yes, I would love to do that.

Reply:
All right, so let's get that one done first nad then we can go back to the Modal.